### PR TITLE
Fix file upload related stuff to reflect reality

### PR DESCRIPTION
**Pull request by newhouse**
## Description of the change

Recent updates to the API around Uploads in GraphQL were not really reflected here. Let's get them up to date. 

Mainly this was around how we used to "wrap" files in an envelope to mimic what Apollo GraphQL Upload was doing (and our backend liked/expected it). That has now changed, so no more wrapper.

However, this still required an alternate way for users to comply with "non-common" streams in our GraphQL operations by providing filename, mimetype and other options to the `form-data.append` call...so I came up with a solution there as well.

Also got the `base64` upload stuff in line with the backend, and tested it out.

Also got to remove a few dependencies...yay.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

Fixes #1

### Development

- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development
- [x] No previous tests unrelated to the changed code fail in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [ ] At least one reviewer has been requested
- [ ] Changes have been reviewed by at least one other engineer
- [ ] The relevant project board has been selected in Projects to auto-link to this pull request

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -296,26 +292,82 @@ describe('Anvil API Client', function () {
...
+            }))
+
+            it('creates a FormData and appends the files map', function () {
+              client.requestGraphQL({ query, variables: $.variables }, clientOptions)
+            })
+          })
+
+          context('file is a Stream', function () {
+            def('variables', () => {
+              const file = fs.createReadStream(path.join(assetsDir, 'dummy.pdf'))
```
Comment (benogle) [inline test/index.test.js:328]: Does this provide the filename / mime? Will this fail asking for a filename / mime when a stream without that info is provided? I assume it would be specified just like the base64 one below?

File: src/index.js [hunk 2]
```diff
@@ -115,14 +126,10 @@ class Anvil {
...
-        generateEtchSignURL,
-      },
-    } = data
+
     return {
       statusCode,
-      url: generateEtchSignURL,
+      url: data?.data?.generateEtchSignURL,
       errors,
     }
```
Comment (Winggo) [inline src/index.js:134]: Snuck in a tiny reformatting change here so that a potential error wouldn't be thrown. Doesn't affect rest of code. Hope that's fine with you @newhouse! 🙂

Comment (newhouse) [reply src/index.js:134]: All good, thx

File: test/index.test.js (hunk 1 above)
Comment (newhouse) [reply test/index.test.js:328]: Great question, but no we are ok. As I [wrote in our docs](https://github.com/anvilco/node-anvil/pull/20/files#diff-04c6e90faac2675aa89e2176d2eec7d8R131), `form-data` will be able to automatically suss out the filename and mime stuff from several "common" types of streams that it deals with:
> Form-Data can recognize and fetch all the required information from common types of streams (fs.readStream, http.response and mikeal's request), for some other types of streams you'd need to provide "file"-related information manually

What happens is that it finds all that info in the "stream"-like thing's properties and it ends up in the multipart form head junk...and on the server side, `GraphQL Upload` will leverage it.

Our API supports `form-data` API's allowance for the manually passing/specifying of filename and mime stuff for these "non common" scenarios via the `options` param that I just referenced here, so users are good to go for all scenarios.

Basically, if `form-data` supports it (which is the only way we can support it), then we can also support it...but sometimes the User has to provide the `options` param depending on what stream-like thing they're going to use.

I've tried this against the API for reals and logged out a bunch of scenarios and it's all good.

The base64 case is obviously different b/c from the framework's point of view this is just your average JSON with some string value.

File: src/index.js (hunk 2 above)
Comment (newhouse) [reply src/index.js:134]: Look at you with your optional chaining!!!

Comment (benogle) [reply src/index.js:134]: yo dawg i heard you like data

**Comment by newhouse** [comment]
good go go @benogle ?

**Review by benogle — APPROVED** [review]
